src/validation_workflow/validation.py

```diff
@@ -6,26 +6,43 @@
 # compatible open source license.
 
 
+import logging
+import shutil
 from abc import ABC, abstractmethod
 from typing import Any
 
+from validation_workflow.download_utils import DownloadUtils
 from validation_workflow.validation_args import ValidationArgs
 
 
 class Validation(ABC):
     """
-    Abstract class for all types of artifact validation
+        Abstract class for all types of artifact validation
     """
 
     def __init__(self, args: ValidationArgs) -> None:
         super().__init__()
         self.args = args
 
+    def check_url(self, url: str) -> bool:
+        if DownloadUtils().download(url, self.tmp_dir) and DownloadUtils().is_url_valid(url):  # type: ignore
+            logging.info(f"Valid URL - {url} and Download Successful !")
+            return True
+        else:
+            raise Exception(f"Invalid url - {url}")
+
+    def copy_artifact(self, filepath: str, tempdir_path: str) -> bool:
+        if filepath:
+            shutil.copy2(filepath, tempdir_path)
+            return True
+        else:
+            raise Exception("Provide local file path")
```

Comment:
I think this exception should be more generic. Maybe 'Path does not exist' or not found?